libraries/AP_Motors/AP_Motors_Class.cpp

```diff
@@ -208,6 +208,21 @@ void AP_Motors::set_limit_flag_pitch_roll_yaw(bool flag)
     limit.yaw = flag;
 }
 
+// returns true if the configured PWM type is digital and should have fixed endpoints
+bool AP_Motors::is_digital_pwm_type() const
+{
+    switch (_pwm_type) {
+        case PWM_TYPE_DSHOT150:
+        case PWM_TYPE_DSHOT300:
+        case PWM_TYPE_DSHOT600:
+        case PWM_TYPE_DSHOT1200:
+            return true;
+        default:
+            break;
+    }
+    return false;
```

Comment:
We should enumerate all values here so when someone adds a pwm type they're forced to evaluate if it should be added here.

Reply:
Similarly elsewhere.

Reply:
done for the new ones, the moved is_dshot_protocol remains with a default.